From 97b563d9c0cf09897c1e7b793ef0fe677c66269a Mon Sep 17 00:00:00 2001
From: Nipun Gupta <nipun.gupta@nxp.com>
Date: Wed, 30 Jan 2019 19:18:13 +0530
Subject: [PATCH] sdk: arm64: dts: reduce usdpaa memory to 4K for LS1046/43

This patch reducing the USDPAA reseved memory to 4K.
In case USDPAA is to be used, 256MB needs to be reserved in the DTS file.

Signed-off-by: Nipun Gupta <nipun.gupta@nxp.com>
---
 arch/arm64/boot/dts/freescale/fsl-ls1043a-rdb-usdpaa.dts | 9 +++++++--
 arch/arm64/boot/dts/freescale/fsl-ls1046a-rdb-usdpaa.dts | 9 +++++++--
 2 files changed, 14 insertions(+), 4 deletions(-)

--- a/arch/arm64/boot/dts/freescale/fsl-ls1043a-rdb-usdpaa.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1043a-rdb-usdpaa.dts
@@ -113,11 +113,16 @@
 		#size-cells = <2>;
 		ranges;
 
+		/* For legacy usdpaa based use-cases, update the size and
+		   alignment parameters. e.g. to allocate 256 MB memory:
+		   size = <0 0x10000000>;
+		   alignment = <0 0x10000000>;
+		*/
 		usdpaa_mem: usdpaa_mem {
 			compatible = "fsl,usdpaa-mem";
 			alloc-ranges = <0 0 0x10000 0>;
-			size = <0 0x10000000>;
-			alignment = <0 0x10000000>;
+			size = <0 0x1000>;
+			alignment = <0 0x1000>;
 		};
 	};
 };
--- a/arch/arm64/boot/dts/freescale/fsl-ls1046a-rdb-usdpaa.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1046a-rdb-usdpaa.dts
@@ -106,11 +106,16 @@
 		#size-cells = <2>;
 		ranges;
 
+		/* For legacy usdpaa based use-cases, update the size and
+		   alignment parameters. e.g. to allocate 256 MB memory:
+		   size = <0 0x10000000>;
+		   alignment = <0 0x10000000>;
+		*/
 		usdpaa_mem: usdpaa_mem {
 			compatible = "fsl,usdpaa-mem";
 			alloc-ranges = <0 0 0x10000 0>;
-			size = <0 0x10000000>;
-			alignment = <0 0x10000000>;
+			size = <0 0x1000>;
+			alignment = <0 0x1000>;
 		};
 	};
 };
